tensorflow/python/kernel_tests/nn_ops/pooling_ops_3d_test.py

```diff
@@ -138,19 +149,21 @@ def testAvgPool3dSamePaddingDifferentStrides(self):
   def testAvgPool3dGrad(self):
     with self.assertRaises(
         (errors.ResourceExhaustedError, errors.InvalidArgumentError)):
-      with self.cached_session():
-        orig_input_shape = constant_op.constant(
-            1879048192, shape=[5], dtype=dtypes.int32)
-        grad = constant_op.constant(
-            1, shape=[1, 3, 2, 4, 2], dtype=dtypes.float32)
-        t = gen_nn_ops.AvgPool3DGrad(
-            orig_input_shape=orig_input_shape,
-            grad=grad,
-            ksize=[1, 1, 1, 1, 1],
-            strides=[1, 1, 1, 1, 1],
-            padding="SAME",
-            data_format="NDHWC")
-        self.evaluate(t)
+      use_gpu = test.is_gpu_available(cuda_only=True)
+      for dtype in self._DtypesToTest(use_gpu):
```

Comment:
Why do we exclude `bfloat16` on GPU? I think we should test both `float32` and `bfloat16` on both CPU and GPU. (If so, we can probably just use a list here. We don't need to create `_DtypesToTest` if it's only used once.)

Reply:
Thanks for the review @penpornk . You have valid suggestions, i addressed them all and pushed another commit.